sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/PathsHelper.java

```diff
@@ -63,14 +63,16 @@ public static String generatePathForNameBased(Resource resourceType, String reso
         throw new IllegalArgumentException(errorMessage);
     }
 
-    private static String generatePathForNameBased(ResourceType resourceType, String resourceFullName, boolean isFeed) {
+    private static String generatePathForNameBased(ResourceType resourceType, String resourceFullName, boolean isFeed, OperationType operationType) {
         if (isFeed && Strings.isNullOrEmpty(resourceFullName) && resourceType != ResourceType.Database) {
             String errorMessage = String.format(RMResources.UnexpectedResourceType, resourceType);
             throw new IllegalArgumentException(errorMessage);
         }
 
         String resourcePath = null;
-        if (!isFeed) {
+        if (resourceType == ResourceType.PartitionKey && operationType == OperationType.Delete) {
+            resourcePath = resourceFullName + "/" + Paths.OPERATIONS_PATH_SEGMENT + "/" + Paths.PARTITION_KEY_DELETE_PATH_SEGMENT;
+        } else if (!isFeed) {
```

Comment:
I would keep the isFeed condition first , as it has separate logic then rest of all, easier to read

Reply:
Done